File: laocai-bot-spring-boot-starter/src/main/java/indi/kyson/laocai/bot/model/segment/TextSegment.java
package indi.kyson.laocai.bot.model.segment;

import com.fasterxml.jackson.annotation.JsonIgnore;

import java.util.Objects;

/**
 * 纯文本消息段。
 */
public final class TextSegment implements Segment {

    record Data(String text) {
        TextSegment toSegment() {
            return new TextSegment(this);
        }
    }

    private final Data data;

    private TextSegment(Data data) {
        this.data = data;
    }

    public static TextSegment of(String text) {
        return new TextSegment(new Data(Objects.requireNonNull(text)));
    }

    @Override
    public String getType() {
        return "text";
    }

    @Override
    public Data getData() {
        return data;
    }

    @JsonIgnore
    public String getText() {
        return data.text();
    }
}

File: laocai-bot-spring-boot-starter/src/main/kotlin/indi/kyson/laocai/bot/segment/TextSegment.kt
package indi.kyson.laocai.bot.segment

import com.fasterxml.jackson.annotation.JsonIgnore

/**
 * 纯文本消息段。
 */
data class TextSegment private constructor(override val data: Data) : Segment {

    data class Data(val text: String) {
        fun toSegment(): TextSegment = TextSegment(this)
    }

    override val type: String
        get() = "text"

    @get:JsonIgnore
    val text: String
        get() = data.text

    companion object {
        @JvmStatic
        fun of(text: String): TextSegment = TextSegment(Data(text))
    }
}
